verifier/src/qe_identity.rs

```diff
@@ -130,14 +132,33 @@ impl QeIdentity {
         let xfrm = u64::from_le_bytes(bytes);
         sgx_attributes_t { flags, xfrm }.into()
     }
+
+    fn verify_time(&self, time: DateTime) -> Result<&Self, Error> {
+        let issue_date = self.issue_date.parse::<DateTime>()?;
+        let next_update = self.next_update.parse::<DateTime>()?;
+        if time < issue_date {
+            Err(Error::QeIdentityNotYetValid)
+        } else if time >= next_update {
+            Err(Error::QeIdentityExpired)
+        } else {
+            Ok(self)
+        }
+    }
+    fn verify_version(&self) -> Result<&Self, Error> {
```

Comment:
Once I added this `verify_version` I was either parsing `QeIdentity` twice inside of `SignedQeIdentity::verify()` or passing the parsed `QeIdentity` to these methods.
These methods no longer needed a reference to `SignedQeIdentity` so it seemed more appropriate to move them into here, `QeIdentity`.